yt/yt/server/master/node_tracker_server/node_tracker.cpp

```diff
@@ -1053,18 +1068,45 @@ class TNodeTracker
             }
         }
 
-        TRack* oldNodeRack = nullptr;
+        TRack* rack = nullptr;
 
         auto* node = FindNodeByAddress(options.DefaultAddress);
         auto isNodeNew = !IsObjectAlive(node);
         if (!isNodeNew) {
             KickOutPreviousNodeIncarnation(node, options.DefaultAddress);
-            oldNodeRack = node->GetRack();
+            rack = node->GetRack();
+        }
+
+        if (options.Rack) {
```

Comment:
Somewhere around here we should probably validate that if `options.DataCenter` is specified then `options.Rack` is specified, too. (I get that this is validated by the node's config but configs change, and it won't hurt to double check things here.)